src/webapi/shadow_root.rs

```diff
@@ -0,0 +1,70 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webcore::reference_type::ReferenceType;
+use webapi::element::Element;
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::node::{INode, Node};
+use webapi::document_fragment::DocumentFragment;
+
+/// The mode associated to a shadow root.
+/// Mainly used in [IElement::attach_shadow](trait.IElement.html#method.attach_shadow) and
+/// [IShadowRoot::mode](trait.IShadowRoot.html#method.mode).
+// https://dom.spec.whatwg.org/#shadowroot-mode
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ShadowRootMode {
+    Open,
+    Closed,
+}
+
+impl ShadowRootMode {
+    pub fn as_str(&self) -> &'static str {
```

Comment:
We probably don't need this public. (So either make it private or `pub(crate)` if it's used outside of this module.)